Document ongoing maintenance section of contribution guide (#727)
submariner-io/submariner-website: src/content/development/contribution-guide/_index.en.md

### Unplanned Work

> 

Reviewer: We should add a short explanation that any task that isnt maintenance needs to be triaged and added to the backlog.

Also we should probably address PRs coming from members not in the Submariner project on GH

#### Ongoing Maintenance

Tasks that relate to ongoing maintenance (version bumps, image trimmings, CI and automation) will usually be small, unplanned tasks that

> will typically occur each release cycle. The associated GitHub issue should be labeled as "maintenance".

Reviewer 2: ```suggestion
 typically occur throughout each release cycle. The associated GitHub issue should be labeled as "maintenance".
```

> typically occur throughout each release cycle. The associated GitHub issue / PR should be labeled as "maintenance".

Reviewer: So a new label?

We already have `automation` label which fits at least the CI/automation stuff. Also version bumps in GHA can use this label.

Generally I think we need a label if it adds value otherwise we're just creating a plethora of them which dont add any real value.

Reviewer 2: I think `maintenance` is meant to be a generalization-of/better-word-for `automation`, which makes sense. I think I was the main person using that label, and I've switched to `maintenance` already.

Reviewer: I am also using `automation` but I'm fine if we replace it with `maintenance`, as long as we don't have two which would be just confusing IMHO :)

Reviewer: I'm replacing all `automation` labels with `maintenance` for open issues and will delete the `automation` label so theres no confusion

> In case the task has a major effect (e.g. needs more than a day of work) then it should go through the exception process.

Reviewer 2: ```suggestion
If the task has a major effect (e.g. needs more than a day of work), then it should go through the exception process.
```

> If the task has a major effect (e.g. needs more than a day of work), then it should go through the exception process.

Reviewer: Since we decided on "prioritized backlog" this makes little to no sense..

If anything perhaps "a task should be reported and triaged".

Also "a day" is quite arbitrary and really depends how much you segment it.

I'd leave this time restriction out altogether as usually these types of tasks arent a pain point for us.

Reviewer 2: Yeah I think the time guideline is a bit arbitrary, not worth trying to specify.

Author: whit that the exception process will be for epics only

Reviewer: > whit that the exception process will be for epics only

You say it as if it's a bad thing..

What's the problem with that?

### Unplanned Work

#### Ongoing Maintenance

Tasks that relate to ongoing maintenance (version bumps, image trimmings, CI and automation) will usually be small and unplanned
tasks that we can't plan ahead for, but we know there will always be work to do.

> Use label "maintenance" on the issue.

Reviewer 3: ... and on PRs? (I’m thinking of dependabot PRs for example where we won’t have an issue filed before the PR is submitted.)

Author: We can use this label for PRs as well

> Use label "maintenance" on the issue.

Reviewer 2: We should make sure this label exists on all repos.

Reviewer 2: I added the `maintenance` label to all repos.

Reviewer 2: I added the `maintenance` label to all repos.

Author: Thanks @dfarrell07

> Use label "maintenance" on the issue.

Reviewer 2: We should make sure this label exists on all repos.

> Tasks that relate to ongoing maintenance (version bumps, image trimmings, CI and automation) will usually be small and unplanned
> tasks that we can't plan ahead for, but we know there will always be work to do.
> Use label "maintenance" on the issue.

Reviewer 4: ```suggestion
Tasks that relate to ongoing maintenance (version bumps, image trimmings, CI and automation) will usually be small, unplanned tasks that will typically occur each release cycle. The associated GitHub issue should be labeled as "maintenance".
```

> In case the task has a major effect (e.g. needs more than a few days of work) then it should go through an exception process.

Reviewer 3: ```suggestion
In case the task has a major effect (e.g. needs more than a few days of work) then it should go through the exception process.
```

I would suggest even a single day of work; anything more than that should go through prioritisation. There might be “short-circuiting” justifications (*e.g.* security issues actually affecting Submariner) but they still need to be made explicit.

### Unplanned Work

During the work on a release, there will be tasks which the team have to add to the projects which was unknown at the planning phase.
There are 3 type of those tasks:

* [Bugs](#bugs).

> * Ongoin maintenance.

Reviewer: ```suggestion
* Ongoing maintenance.
```

> * New [Epics](#epics) or small independent tasks that need to start work on.

Reviewer: ```suggestion
* New [epics](#epics) or independent [tasks](#tasks).
```

> Bugs do not need to be part of a release planning, and the two other types are listed below:

Reviewer: ```suggestion
The team triages all tasks and bugs as part of a weekly meeting.
The bugs can be worked on immediately, while other types of issues are described below.
```

> #### Ongoing Maintenance

Reviewer: I wonder if it would make more sense to put this under the "tasks, bugs, epics" since it's not strictly related to planning or not, its like bugs that are continuous.

Author: Which part do you consider moving to "tasks, bugs, epics"?
The first ("tasks, bugs, epics") is a definition, here is in relation to the  work process

Author: @mkolesnik - what do suggest to do here?

Reviewer: Just the general description of this type of task.

I'd put it under `tasks` sub-section in "tasks, bugs, epics" section.

Tasks that relate to ongoing maintenance (version bumps, image trimmings, CI and automation) will usually be small, unplanned tasks that
typically occur throughout each release cycle. The associated GitHub issue / PR should be labeled as "maintenance".

#### Exception Process

> In some cases, a task or an epic will need to be worked on in the middle of a release cycle.

Reviewer: ```suggestion
In some cases, a task or an epic will be reported in the middle of a release cycle.
Small and medium sized tasks will be triaged and prioritized in the same manner as other issues.
Once triaged and prioritized, small and medium tasks can be worked on according to their priority.
```

> Any task should be reported and triaged in the backlog board.

Reviewer: ```suggestion
```

All issues (including bugs) need to be triaged so I added this information in the top level section

> For an epic, an enhancement proposal need to be draft and merged in the same manner for all other epics.
> Bring it up to discussion in the weekly community meeting. It's up to the proposer to convince why this issue takes precedence
> over pre-planned issues. The epic should get 50% approval to be accepted. A label "unplanned" should be added to the epic.

Reviewer: ```suggestion
Epics would need to undergo an exception process.
The epic's proponent would need to bring it up to discussion in the weekly community meeting.
It's up to them to convince why this issue takes precedence over other pre-planned and triaged issues.
A 50% approval from the project team is required for it to be accepted in the middle of the release.

If the epic is approved during the exception process:

* The team might de-scope other tasks or epics.
* A label `unplanned` should be added to it.
* It would require an enhancement proposal to be drafted and reviewed in the same manner as all other epics.
* If the enhancement proposal is accepted and merged, work on the epic can begin.
```

### Unplanned Work

> During the work on a release, there will be tasks which the team have to add to the projects which was unknown at the planning phase.

Reviewer 2: ```suggestion
During a release, new work that was unknown during the planning phase will emerge.
```

### Unplanned Work

During the work on a release, there will be tasks which the team have to add to the projects which was unknown at the planning phase.

> There are 3 type of those tasks:

Reviewer 2: ```suggestion
This work is typically one of three types:
```

* [Bugs](#bugs).
* Ongoing maintenance [tasks](#tasks).
* New [epics](#epics) or independent [tasks](#tasks).

> The team triages all tasks and bugs as part of a weekly meeting.

Reviewer 2: ```suggestion
The team [triages tasks and bugs as part of a weekly meeting]https://docs.google.com/document/d/1ICvOBOUpIEI45Yqcjf26Z1e5mZNzYsEN-H7_3xx54jI).
```

The bugs can be worked on immediately, while other types of issues are described below.

#### Exception Process

In some cases, a task or an epic will be reported in the middle of a release cycle.
Small and medium sized tasks will be triaged and prioritized in the same manner as other issues.
Once triaged and prioritized, small and medium tasks can be worked on according to their priority.

Epics would need to undergo an exception process.
The epic's proponent would need to bring it up to discussion in the weekly community meeting.

> It's up to them to convince why this issue takes precedence over other pre-planned and triaged issues.

Reviewer 2: ```suggestion
It's up to the proponent to show why the new issue should take precedence over other pre-planned and triaged issues.
```

> A 50% approval from the project team is required for it to be accepted in the middle of the release.

Reviewer 2: ```suggestion
A 50% approval from the Project Owners is required for it to be accepted in the middle of the release.
```

Author: I think in this case the team needs to agree, not only project owners

Reviewer 2: What is the "project team"? I don't think that defines any clear group of people. Also, the Project Owners are the ones that ultimately choose the technical direction of the project and guide releases. Please see https://submariner.io/community/contributor-roles/.

Author: @mkolesnik - since this phrase was your suggestion, do you think we should change this to `Project Owners`?
@dfarrell07 - the approval process should be done during the weekly community meeting, hence `project team` is considered as the meeting participants

Reviewer: We can change to "project members" if you want to reflect the community aspect.

I also think "project committers" can be fine if we want a stricter list.

IMHO "project owners" is too narrow as it only has 3 people in actuality, and I dont see why committers cant cast a vote in these decisions.

Reviewer 2: I think "Project Committers" is fine as it's a well-defined group and makes sense in practice. Or maybe "Committers to the files expected to be impacted".

But to be clear for the record, this is actually delegating a part of a responsibility that's ultimately granted to Project Owners by Submariner's governance.

Owner Responsibilities and Privileges:
* Make and approve technical design decisions for the project
* Set technical direction and priorities for the project
* Define milestones and releases

https://submariner.io/community/contributor-roles/#owner-responsibilities-and-privileges

Of course it's fine and normal to delegate responsibilities to spread out work to the most relevant stakeholders. Just want to be clear that they are already defined as the ultimate authority in this domain.

Reviewer 2: I'm also okay with "Project Members", but I'm not sure it would be a good use of the time of that large group of lightly-connected-to-SubM people. I think we'd have low voter turnout or have to spam folks to ask them to spend time on something that they likely don't really care about.

Reviewer 2: ```suggestion
A 50% approval from the [Project Members](https://submariner.io/community/contributor-roles/#member) in attendance at the meeting or having voted before the meeting is required for an epic to be accepted in the middle of the release.
```

If the epic is approved during the exception process:

* The team might de-scope other tasks or epics.

> * A label `unplanned` should be added to it.

Reviewer 2: What's the reason for this?

Author: if we want an easy way to compare how much of the work done was planned in the release planning vs. the worked we added after the original plan, a label will be usefull
